easybuild/easyblocks/r/raxml.py

```diff
@@ -0,0 +1,194 @@
+            # check for supported labels in optarch (by given priority order)
+            for instruct_set in unfilter_features:
+                if instruct_set in optarch:
+                    dbg_msg = "Restricting supported features in RAxML to '%s' per optarch: %s"
+                    self.log.debug(dbg_msg, instruct_set, optarch)
+                    return [instruct_set]
+            # anything else is unsupported
+            if optarch:
+                print_warning("Unsupported 'optarch' configuration setting for RAxML, ignoring: %s" % optarch)
+
+            return support_features
+
+        def has_cpu_feature(feature):
+            """
+            Check if host CPU architecture has given feature
+            """
+            if HAVE_ARCHSPEC:
+                import archspec.cpu
+
+                host = archspec.cpu.host()
+                return feature.lower() in host
+
+            try:
+                return any(f in get_cpu_features() for f in RAXML_CPU_FEATURES[feature])
+            except KeyError:
+                raise EasyBuildError("Unknown CPU feature level for RAxML: %s", feature)
+
+        def list_filename_variants(main_features, extra_features, prefix, suffix, divider):
+            """
+            Returns list of RAxML filenames for the combination of all given features
+            """
+            if not isinstance(main_features, list):
+                main_features = list(main_features)
+            if not isinstance(extra_features, list):
+                extra_features = list(extra_features)
+
+            # Permutations of features
+            all_features = [(mf, xf) for mf in main_features for xf in extra_features]
+
+            # Prepend/Append prefix/suffix
+            file_variants = [(prefix,) + variant + (suffix,) for variant in all_features]
+            file_variants = [divider.join([segment for segment in variant if segment]) for variant in file_variants]
+
+            return file_variants
+
+        # Filter target CPU instructions sets according to optarch
+        cpu_features = filter_optarch_features(RAXML_INSTRUCT_SETS)
+
+        if len(cpu_features) > 1:
+            # unrestricted optimization settings, set optimization level for host micro-architecture
+            cpu_features = [feat for feat in cpu_features if has_cpu_feature(feat)]
+            dbg_msg = "Enabling the following CPU optimizations for RAxML by autodetection: %s"
+            self.log.debug(dbg_msg, ", ".join(cpu_features))
+            # add generic build
+            cpu_features.append(None)
+
+        # Set parallelization level of RAxML for current toolchain
+        parallel_features = RAXML_PARALLEL_FEATURES["nompi"]
+        if self.toolchain.options.get("usempi", None):
+            parallel_features.extend(RAXML_PARALLEL_FEATURES["mpi"])
+
+        # List of builds to carry out
+        self.target_makefiles = list_filename_variants(cpu_features, parallel_features, "Makefile", "gcc", ".")
```

Comment:
Is always using `gcc` OK? What if we're installing `RAxML` with Intel compilers?